main.sh

```diff
@@ -22,8 +22,10 @@ detect_build_necessity() {
   IFS=' ' read -r -a keywords <<< "$keywords_str"
 
   case $commit_msg in
-    --|' ') commit_msg="$(git log --format=%B -n 1 HEAD)" ;;
-    *) log_warning "commit_msg is set as:  '$commit_msg'" ;;
```

Comment:
warning should be shown when commit_msg not passed instead. Not other way round.